### PR TITLE
Add test to make sure `/messages` behaves as expected for non-existent `room_id`'s

### PR DESCRIPTION
**Pull request by MadLittleMods**
Add test to make sure `/messages` behaves as expected for non-existent `room_id`'s

Tests for Synapse regression fix: https://github.com/matrix-org/synapse/pull/12683
Issue: https://github.com/matrix-org/synapse/issues/12678

### CODE REVIEW

File: tests/csapi/room_messages_test.go [hunk 1]
```diff
@@ -59,6 +59,25 @@ func TestSendAndFetchMessage(t *testing.T) {
 	})
 }
 
+// With a non-existent room_id, GET /rooms/:room_id/messages returns 403
+// forbidden ("You aren't a member of the room").
```
Comment (MadLittleMods) [inline tests/csapi/room_messages_test.go:63]: As described in the spec, 

> 403 | You aren’t a member of the room.
>
> *-- https://spec.matrix.org/latest/client-server-api/#get_matrixclientv3roomsroomidmessages*

File: tests/csapi/room_messages_test.go [hunk 2]
```diff
@@ -59,6 +59,24 @@ func TestSendAndFetchMessage(t *testing.T) {
...
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+	roomID := "!does-not-exist:hs1"
+
+	// then request messages from the room
+	queryParams := url.Values{}
+	queryParams.Set("dir", "b")
+	res := alice.DoFunc(t, "GET", []string{"_matrix", "client", "r0", "rooms", roomID, "messages"}, client.WithQueries(queryParams))
+	must.MatchResponse(t, res, match.HTTPResponse{
+		StatusCode: http.StatusForbidden,
```
Comment (MadLittleMods) [inline tests/csapi/room_messages_test.go:76]: Looks like Dendrite doesn't match the spec either at the moment. Should we skip for now?

> `room_messages_test.go:75: MatchResponse got status 500 want 403 - http://localhost:49256/_matrix/client/r0/rooms/%21does-not-exist:hs1/messages?dir=b => {"errcode":"M_UNKNOWN","error":"Internal Server Error"}`
>
> *-- https://github.com/matrix-org/complement/runs/6362103594?check_suite_focus=true#step:8:632*

Comment (kegsay) [reply tests/csapi/room_messages_test.go:76]: Wait a bit and we'll get a Dendrite fix in :)

**Comment by kegsay** [comment]
Silly github auto-closed the PR when I merged the Dendrite fix :S

**Review by kegsay — APPROVED** [review]

File: tests/csapi/room_messages_test.go (hunk 2 above)
Comment (MadLittleMods) [reply tests/csapi/room_messages_test.go:76]: https://github.com/matrix-org/dendrite/pull/2450 🎉

**Comment by MadLittleMods** [comment]
Thanks for the review and merge @kegsay! Cool to see the Dendrite fix for this to all align as well! 🦏

**Pull request merged**